dcs-packaging-tool/dcs-packaging-tool-api/src/main/java/org/dataconservancy/packaging/tool/api/DomainProfileService.java

```diff
@@ -0,0 +1,163 @@
+package org.dataconservancy.packaging.tool.api;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import org.dataconservancy.packaging.tool.model.dprofile.DomainProfile;
+import org.dataconservancy.packaging.tool.model.dprofile.NodeTransform;
+import org.dataconservancy.packaging.tool.model.dprofile.PropertyType;
+import org.dataconservancy.packaging.tool.model.dprofile.PropertyValue;
+import org.dataconservancy.packaging.tool.model.ipm.Node;
+
+/**
+ * Service to perform operations on a tree of domain objects using a domain
+ * profile.
+ */
+public interface DomainProfileService {
```

Comment:
The idea is that a DomainProfileService implementation would be able to operation on any DomainProfile.